doc/devel/development.md

```diff
@@ -0,0 +1,43 @@
+# Developer guide 
+This document explains how to setup your dev environment. 
+
+## Prerequisite 
+- kubernetes 1.4 
+- go-lang version 1.7
+- docker 
+- glide 
+
+## Setup
+### Step 1
```

Comment:
On a second thought, I think we have documented how to build stuff in "hack/release". Maybe we can just add a link to it.

Other than build, we can also talk about code style, how to test in this doc. We also have a doc about e2e already.